common-kafka/src/main/java/com/cerner/common/kafka/consumer/ConsumerOffsetClient.java

```diff
@@ -143,9 +143,14 @@ public ConsumerOffsetClient(Consumer<Object, Object> consumer) {
                 .stream().collect(Collectors.toMap(Function.identity(), s -> time));
         Map<TopicPartition, OffsetAndTimestamp> foundOffsets = consumer.offsetsForTimes(topicTimes);
 
+        //Remove the entry from the foundOffsets map if the offset does not exist.
+        foundOffsets.entrySet().removeIf(e -> e.getValue() == null);
+
         //merge the offsets together into a single collection.
         Map<TopicPartition, Long> offsets = new HashMap<>();
-        offsets.putAll(foundOffsets.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().offset())));
+        offsets.putAll(foundOffsets.entrySet()
+                .stream()
```

Comment:
Because the `foundOffsets` map is not guaranteed to be a mutable implementation, we should instead use `.filter` here to exclude entries with a null value.

Reply:
So the problem doing that was: 
The missingPartitions in the end look for entries in foundOffsets. So given that we are not going to remove the null offsets from foundOffsets it won't add the endOffset for the missingOnes right?

Reply:
I think you can just change foundOffsets -> offsets there on line 157.

Reply:
updated